### ISSUE TITLE
PCF-6: URLs - interaction between routing and query parameters

### ISSUE THREAD

**Issue by CuriousCactus** [prophet-kn/dnd_spells_book#63]
**Describe the bug**
The routing doesn't play nicely with the spell/feat pinning, leading to some weird URLs:
https://prophet-kn.github.io/dnd_spells_book/?s_id=466,467#/feats-table

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'List of Spells'
2. Pin some spells
3. Go to 'List of Feats'
4. See strange URL: https://prophet-kn.github.io/dnd_spells_book/?s_id=466,467#/feats-table

**Expected behaviour**
I expect the URL to be composed like this:
https://prophet-kn.github.io/dnd_spells_book/feats-table?s_id=466,467

The behaviour actually seems to be fine, it's just the ordering of the URL that's odd.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20
 - Browser: Firefox
 - Version: 78.0.2 (64-bit)

**Comment by prophet-kn** [comment]
Yeah, thing is, we cannot use normal react-router's Route because this is hosted on github pages and the way they work with different paths makes it print 404, especially cus this is react. That is why we need to use hash route. Funnily enough the pins still stay, the URL just looks funny. This can probably be solved by tweaking the code that appends the query params, to read the whole path not just do whatever it wants.
